Userland/BuggieBox/main.cpp

```diff
@@ -46,12 +46,29 @@ ENUMERATE_UTILITIES(DECLARE_ENTRYPOINT)
 
 static void fail()
 {
-    out(stderr, "Direct running of BuggieBox was detected without finding a proper requested utility.\n");
-    out(stderr, "The following programs are supported: uname, env, lsblk, file, df, mount, umount, mkdir, ");
-    out(stderr, "rmdir, rm, chown, chmod, cp, ln, ls, mv, cat, md5sum, sha1sum, sha256sum, sha512sum, sh, uniq, id, tail, ");
-    out(stderr, "find, less, mknod, ps\n");
-    out(stderr, "To use one of these included utilities, create a symbolic link with the target being this binary, and ensure the basename");
-    out(stderr, "is included within.\n");
+#define TOOL_NAME(name) #name,
+    auto const tool_names = {
+        "md5sum", "sha1sum", "sha256sum", "sha512sum",
+        "Shell", ENUMERATE_UTILITIES(TOOL_NAME)
+    };
+#undef TOOL_NAME
+
+    outln(stderr);
+    outln(stderr, "Usage:");
+    outln(stderr, "1. Create a symbolic link with the target being this binary,");
+    outln(stderr, "   and ensure the basename is one of the supported utility's name.");
+    outln(stderr, "2. Specify a utility as an argument:");
+    outln(stderr, "   $ BuggieBox UTILITY");
+
+    outln(stderr);
+    outln(stderr, "The following utilities are supported:");
+    int count = 0;
+    for (auto const& tool_name : tool_names) {
```

Comment:
Do we want to sort these alphabetically before printing? I feel like it would be a lot nicer than having all the aliased utilities at the beginning or end.

Alternatively, we could modify `ENUMERATE_UTILITIES` to also include aliases and then the caller decides what it wants to do with either type.

Reply:
How about `ENUMERATE_ALIASES` which takes two arguments: 1. the alias name and 2. the main function name (`checksum_main` or `sh_main`)?

Reply:
I imagined something like the following:

```cpp
#define ENUMERATE_UTILITIES(U, A) \
    [...]
    U(lsblk)                      \
    A(md5sum, checksum)           \
    U(mkdir)                      \
    [...]
```

But really, as long as it does what it needs to do and is maintainable, pretty much anything works.

Reply:
Done. I kept the symbolic link info. Sorted statically for now.